packages/web/components/icon/types.ts

```diff
@@ -0,0 +1,8 @@
+export type Icons = "star";
+
+export interface IconProps {
+  color?: string;
+  name?: Icons;
+  onClick?: () => void;
+  size?: number;
+}
```

Comment:
I've shared this convention with jose as well. IMO it's quicker to read the component when props are embedded directly in the type param of the `FunctionComponent` type (equivalently `React.FC`, but we conventionally use `FunctionComponent`). Example `FunctionComponent<{ color?: string, ... }>`